src/coordinator/consent.js

```diff
@@ -0,0 +1,137 @@
+ * @param {string} id UUID
+ * @param {string} text The body of the consent form in HTML
+ * @param {string} updatedAt ISO 8601 Extended Format date/time string
+ * @param {ContxtUser[]} userApproval An array of users. If empty, the user has not consented
+ */
+
+/**
+ * @typedef {Object} ContxtUserConsentApproval
+ * @param {string} consentId
+ * @param {string} createdAt ISO 8601 Extended Format date/time string
+ * @param {string} id UUID
+ * @param {string} updatedAt ISO 8601 Extended Format date/time string
+ * @param {string} userId UUID
+ */
+
+/**
+ * Module for managing application consent
+ *
+ * @typicalname contxtSdk.coordinator.consent
+ */
+class Consent {
+  /**
+   * @param {Object} sdk An instance of the SDK so the module can communicate with other modules
+   * @param {Object} request An instance of the request module tied to this module's audience.
+   * @param {string} baseUrl The base URL provided by the parent module
+   */
+  constructor(sdk, request, baseUrl) {
+    this._baseUrl = baseUrl;
+    this._request = request;
+    this._sdk = sdk;
+  }
+
+  /**
+   * Accepts a user's consent to an application
+   *
+   *
+   * API Endpoint: '/consents/:consentId/accept'
+   * Method: POST
+   *
+   * @param {string} consentId The ID of the consent form the user is accepting
+   *
+   * @returns {Promise}
+   * @fulfill {ContxtUserConsentApproval}
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.coordinator.consent
+   *   .accept('coordinator', '36b8421a-cc4a-4204-b839-1397374fb16b')
+   *   .then((userApproval) => console.log(userApproval))
+   *   .catch((err) => console.log(err));
+   */
+  accept(consentId) {
+    if (!consentId) {
+      return Promise.reject(
+        new Error('A consent ID is required for accepting consent')
+      );
+    }
+
+    const { accessToken } = this._sdk.auth._getStoredSession();
+
+    if (!accessToken) {
+      return Promise.reject(new Error('A valid JWT token is required'));
+    }
+
+    return this._request
+      .post(`${this._baseUrl}/consents/${consentId}/accept`, {
+        access_token: accessToken
+      })
+      .then((userApproval) => toCamelCase(userApproval));
+  }
+
+  /**
+   * Verify if application consent is needed from the user
+   *
+   *
+   * API Endpoint: '/applications/consent'
+   * Method: POST
+   *
+   * @returns {Promise}
+   * @fulfill {ContxtApplicationConsent}
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.coordinator.consent
+   *   .verify('coordinator')
+   *   .then((applicationConsent) => console.log(applicationConsent))
+   *   .catch((err) => console.log(err));
+   */
+  verify() {
+    const { accessToken } = this._sdk.auth._getStoredSession();
+
+    if (!accessToken) {
+      return Promise.reject(new Error('A valid JWT token is required'));
+    }
+
+    return this._request
+      .post(`${this._baseUrl}/applications/consent`, {
+        access_token: accessToken
```

Comment:
This will only ever get the consent tree around whatever app this method is used from -- is this the intent? Can we get some further documentation around this?

Reply:
> is this the intent?

I don't know what the intent is or was for these endpoints, I just learned about these a few days ago when I was told to move it as-is from contxt 1.0 to contxt 2.0. The endpoints remain as they were, the only change is adding them to the sdk. 

I could attempt to document the reasoning behind extracting the active application ID out of the JWT token instead of explicitly asking based on applicaiton ID, but to be honest, I'd most likely be wrong.